apps/library/lib/library/board.ex

```diff
@@ -0,0 +1,56 @@
+defmodule Library.Board do
+
+  @moduledoc """
+  contains functions for manipulating the game board
+  """
+
+  alias Library.Letter
+  @width 5
+  @height 5
+  @number_of_letters @width * @height
+  @generator Library.LetterGenerator
+
+  def generate do
+    @number_of_letters
+    |> @generator.generate
+    |> Stream.with_index
+    |> Enum.map(fn({letter, index}) ->
+      %{id: index, letter: letter, owner: 0, surrounded: false}
+    end)
+  end
+
+  @doc """
+  if all letters have an owner the board is completed and the game is over
+  """
+  def completed?(letters) do
+    Enum.all? letters, fn(letter) -> letter.owner != 0 end
+  end
+
+  @doc """
+  add a word to the board by replacing the owner of the submitted letters
+  """
+  def add_word(word, owner, board) do
```

Comment:
I think board should be the first argument - imo it's like passing the pid around